### PR TITLE
Add Manpage artifact

### PR DESCRIPTION
**Pull request by timsly**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This is a follow-up PR for discussion here - https://github.com/Homebrew/homebrew-cask/pull/70869

It adds the `manpage` artifact, basically a shortcut for moving manpages to the right directory

By default, it tries to read a number from the source string
```ruby
cask 'my-cask' do
  manpage 'manpage.1'
end
```
so the manpage will be moved to `HOMEBREW_PREFIX/"share/man/man1/manpage.1"`

The section can be also specified explicitly 
```ruby
cask 'my-cask' do
  manpage 'manpage.1', section: 3
end
```
so the manpage will be moved to `HOMEBREW_PREFIX/"share/man/man3/manpage.3"`

### CODE REVIEW

File: Library/Homebrew/cask/artifact.rb [hunk 1]
```diff
@@ -23,6 +23,7 @@
 require "cask/artifact/suite"
 require "cask/artifact/uninstall"
 require "cask/artifact/zap"
+require "cask/artifact/manpage"
```
Comment (reitermarkus) [inline Library/Homebrew/cask/artifact.rb]: Put this in alphabetical order.

File: Library/Homebrew/rubocops/cask/constants/stanza.rb [hunk 2]
```diff
@@ -32,6 +32,7 @@ module Constants
           :vst_plugin,
           :artifact,
           :stage_only,
+          :manpage,
```
Comment (reitermarkus) [inline Library/Homebrew/rubocops/cask/constants/stanza.rb]: Move this after `:binary`.

File: Library/Homebrew/cask/dsl.rb [hunk 3]
```diff
@@ -46,6 +46,7 @@ class DSL
       Artifact::Vst3Plugin,
       Artifact::Uninstall,
       Artifact::Zap,
+      Artifact::Manpage,
```
Comment (reitermarkus) [inline Library/Homebrew/cask/dsl.rb]: Also put this in alphabetical order.

File: Library/Homebrew/cask/artifact/manpage.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require "cask/artifact/moved"
+
+require "extend/hash_validator"
+using HashValidator
+
+module Cask
+  module Artifact
+    class Manpage < Moved
```
Comment (reitermarkus) [inline Library/Homebrew/cask/artifact/manpage.rb]: Make this inherit from `Symlinked` instead.

**Comment by reitermarkus** [comment]
Remove everything except the 

```
manpage 'manpage.1'
```

syntax for now. There is no cask that needs the section to be changed currently.

**Comment by timsly** [comment]
@reitermarkus Done

File: Library/Homebrew/cask/artifact/manpage.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+      def initialize(cask, source)
+        super
+      end
+
+      def resolve_target(_target)
+        config.manpagedir.join("man#{section}", target_name)
+      end
+
+      def section
```
Comment (reitermarkus) [inline Library/Homebrew/cask/artifact/manpage.rb]: Remove this in favour of the `@section` variable.

File: Library/Homebrew/cask/artifact/manpage.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    class Manpage < Symlinked
+      def self.from_args(cask, source)
+        section = source.split(".").last
+
+        raise CaskInvalidError, "section should be a positive number" unless section.to_i.positive?
+
+        new(cask, source)
+      end
+
+      def initialize(cask, source)
```
Comment (reitermarkus) [inline Library/Homebrew/cask/artifact/manpage.rb]: Add a `section` argument.

File: Library/Homebrew/cask/artifact/manpage.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require "cask/artifact/symlinked"
+
+module Cask
+  module Artifact
+    class Manpage < Symlinked
+      def self.from_args(cask, source)
+        section = source.split(".").last
```
[the remark this answers is not in the document]
Comment (timsly) [reply Library/Homebrew/cask/artifact/manpage.rb]: 👍 I will also add `|l` as there is also `Section L`

**Comment by timsly** [comment]
Done, With the explicit argument, It looks cleaner

**Comment by timsly** [comment]
@reitermarkus I am wondering about this place https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/artifact/abstract_artifact.rb#L47-L76
Is there a specific position where the `Manpage` should be added?

**Comment by reitermarkus** [comment]
Add it after `Binary`, since it kind of behaves in the same way.

File: Library/Homebrew/cask/artifact/manpage.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      end
+
+      def initialize(cask, source, section)
+        @section = section
+
+        super(cask, source)
+      end
+
+      def resolve_target(_target)
+        config.manpagedir.join("man#{section}", target_name)
```
Comment (reitermarkus) [inline Library/Homebrew/cask/artifact/manpage.rb]: Just use `target` here and remove the `target_name` method below.

**Comment by timsly** [comment]
@reitermarkus Please check. I've addressed all your suggestions

File: Library/Homebrew/test/cask/artifact/manpage_spec.rb [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    let(:install_phase) {
+      lambda do
+        cask.artifacts.select { |a| a.is_a?(described_class) }.each do |artifact|
+          artifact.install_phase(command: NeverSudoSystemCommand, force: false)
+        end
+      end
+    }
+
+    let(:source_path) { cask.staged_path.join("manpage.1") }
+    let(:expected_section) { "" }
```
Comment (reitermarkus) [inline Library/Homebrew/test/cask/artifact/manpage_spec.rb]: I don't think this is needed.

File: Library/Homebrew/test/cask/artifact/manpage_spec.rb [hunk 10]
```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+describe Cask::Artifact::Manpage, :cask do
+  let(:cask_token) { "" }
```
Comment (reitermarkus) [inline Library/Homebrew/test/cask/artifact/manpage_spec.rb]: This is also not needed.

**Comment by timsly** [comment]
Cleaned up specs

**Comment by reitermarkus** [comment]
Please add it to https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/all_stanzas.md.

**Comment by reitermarkus** [comment]
Thanks, @timsly!

**Comment by timsly** [comment]
@reitermarkus Menetioned `manpage` in all_stanzas.md - https://github.com/Homebrew/homebrew-cask/pull/71393

**Pull request merged**